src/redux/slices/questionsListSlice.ts

```diff
@@ -0,0 +1,90 @@
+import { createSlice, PayloadAction, createAsyncThunk } from '@reduxjs/toolkit';
+import { RootState } from '../store';
+import { capitalsQuizData } from '../../config/capitalsQuizData';
+import { riversQuizData } from '../../config/riversQuizData';
+
+interface Question {
+  id: number;
+  question: string;
+  options: string[];
+  difficulty: string;
+}
+
+export interface QuestionsListState {
+  questionsList: {
+    id: number;
+    question: string;
+    options: string[];
+  }[];
+  status: 'idle' | 'loading' | 'succeeded' | 'failed';
+  error: string | null;
+}
+
+const initialState: QuestionsListState = {
+  questionsList: [],
+  status: 'idle',
+  error: null,
+};
+
+function getRandomNumber(): number {
+  return Math.floor(Math.random() * 4);
+}
+
+export const fetchQuestions = createAsyncThunk(
+  'questions/fetchQuestions',
+  async (_, { getState }) => {
+    const state = getState() as RootState;
+    let questions: Question[] = [];
+
+    switch (state.quizConfig.category) {
+      case 'capitals':
+        questions = capitalsQuizData.questions.filter((question: Question) => question.difficulty === state.quizConfig.difficulty).splice(0, state.quizConfig.numQuestions);
+        switch (state.quizConfig.type) {
+          case 'Multiple Choice':
+            questions.map(item => item.question = `What is the capital of ${item.question}?`)
+            break;
+          case 'True/False':
+            questions.map(item => item.question = `Is ${item.options[getRandomNumber()]} the capital of ${item.question}?`)
+            break;
+        }
+        break;
+      case 'rivers':
+        questions = riversQuizData.questions.filter((question: Question) => question.difficulty === state.quizConfig.difficulty).splice(0, state.quizConfig.numQuestions);
```

Comment:
this and the code on the line 41 are the same
maybe you may move them to the declaration of `questions`?